app/sidekiq/evss/disability_compensation_form/submit_form526.rb

```diff
@@ -77,58 +77,61 @@ class SubmitForm526 < Job
-            # submit 526 through Lighthouse API
-            # 1. get user's ICN
-            user_account = UserAccount.find_by(id: submission.user_account_id) ||
-                           Account.find_by(idme_uuid: submission.user_uuid)
-            icn = user_account.icn
-            # 2. transform submission data to Lighthouse format
-            transform_service = EVSS::DisabilityCompensationForm::Form526ToLighthouseTransform.new
-            body = transform_service.transform(submission.form['form526'])
-            # 3. send transformed submission data to Lighthouse endpoint
-            service = BenefitsClaims::Service.new(icn)
-            raw_response = service.submit526(body)
-            # 4. convert raw response from Lighthouse to a FormSubmitResponse for further processing (claim_id, status)
-            raw_response_struct = OpenStruct.new({
-                                                   body: { claim_id: raw_response.body },
-                                                   status: raw_response.status
-                                                 })
-            response = EVSS::DisabilityCompensationForm::FormSubmitResponse
-                       .new(raw_response_struct.status, raw_response_struct)
-          else
-
-            response = service.submit_form526(submission.form_to_json(Form526Submission::FORM_526))
+          begin
+            super(submission_id)
+            submission.prepare_for_evss!
+          rescue => e
+            handle_errors(submission, e)
+            return
+          end
+
+          begin
+            submission.mark_birls_id_as_tried!
+
+            # send submission data to either EVSS or lighthouse
+            if Flipper.enabled?(:disability_compensation_lighthouse_submit_migration)
+              # submit 526 through Lighthouse API
+              # 1. get user's ICN
+              user_account = UserAccount.find_by(id: submission.user_account_id) ||
+                             Account.find_by(idme_uuid: submission.user_uuid)
+              icn = user_account.icn
+              # 2. transform submission data to Lighthouse format
+              transform_service = EVSS::DisabilityCompensationForm::Form526ToLighthouseTransform.new
+              body = transform_service.transform(submission.form['form526'])
+              # 3. send transformed submission data to Lighthouse endpoint
+              service = BenefitsClaims::Service.new(icn)
+              raw_response = service.submit526(body)
+              # 4. convert response from Lighthouse to a FormSubmitResponse for further processing (claim_id, status)
+              raw_response_struct = OpenStruct.new({
+                                                     body: { claim_id: raw_response.body },
+                                                     status: raw_response.status
+                                                   })
+              response = EVSS::DisabilityCompensationForm::FormSubmitResponse
```

Comment:
I'm curious if there's a reason you didn't set the conditional equal to `response` on L104?

Reply:
In hindsight, I should have. I suppose I adopted what already existed without recognizing that improvement.